Google.Authenticator/TwoFactorAuthenticator.cs

```diff
@@ -171,16 +187,43 @@ public string GetCurrentPIN(string accountSecretKey)
             return GeneratePINAtInterval(accountSecretKey, GetCurrentCounter());
         }
         
-         public string GetCurrentPIN(string accountSecretKey,DateTime now)
+        public string GetCurrentPIN(string accountSecretKey, DateTime now)
         {
             return GeneratePINAtInterval(accountSecretKey, GetCurrentCounter(now,_epoch,30));
         }
 
+        public string[] GetCurrentPINs(byte[] accountSecretKey)
+        {
+            return GetCurrentPINs(accountSecretKey, DefaultClockDriftTolerance);
+        }
+
         public string[] GetCurrentPINs(string accountSecretKey)
         {
             return GetCurrentPINs(accountSecretKey, DefaultClockDriftTolerance);
         }
 
+        public string[] GetCurrentPINs(byte[] accountSecretKey, TimeSpan timeTolerance)
+        {
+            List<string> codes = new List<string>();
+            long iterationCounter = GetCurrentCounter();
+            int iterationOffset = 0;
+
+            if (timeTolerance.TotalSeconds > 30)
+            {
+                iterationOffset = Convert.ToInt32(timeTolerance.TotalSeconds / 30.00);
+            }
+
+            long iterationStart = iterationCounter - iterationOffset;
+            long iterationEnd = iterationCounter + iterationOffset;
+
+            for (long counter = iterationStart; counter <= iterationEnd; counter++)
+            {
+                codes.Add(GeneratePINAtInterval(accountSecretKey, counter));
+            }
+
+            return codes.ToArray();
+        }
+
         public string[] GetCurrentPINs(string accountSecretKey, TimeSpan timeTolerance)

```

Comment:
Please change this to convert the string to a byte array and then call your new `GetCurrentPINs` to avoid code duplication.